src/device.rs

```diff
@@ -1384,8 +1393,13 @@ impl<E: StreamSample> RxStream<E> {
                 &mut self.time_ns as *mut _,
                 timeout_us as _,
             ))?;
-
-            Ok(len as usize)
+            // Is it better to return this StreamResult or just make the values public off of rx_stream
```

Comment:
See https://github.com/kevinmehall/rust-soapysdr/issues/6 for prior discussion on this if you haven't already seen it.